plugin/src/main/scala/com/softwaremill/clippy/AdviceLoader.scala

```diff
@@ -10,20 +11,34 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.io.Source
 import scala.tools.nsc.Global
 import scala.util.{Failure, Success, Try}
-import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
 
 class AdviceLoader(global: Global, url: String, localStoreDir: File, projectAdviceFile: Option[File])(implicit ec: ExecutionContext) {
   private val OneDayMillis = 1000L * 60 * 60 * 24
 
   private val localStore = new File(localStoreDir, "clippy.json.gz")
 
+  private def listFilesForFolder(folder: File): List[File] = {
+    folder.listFiles.flatMap { fileEntry =>
+      if (fileEntry.isDirectory)
+        listFilesForFolder(fileEntry)
+      else List(fileEntry)
+    }.toList
+  }
+
   private val resourcesAdvice: List[Advice] =
     getClass.getClassLoader
-      .getResources("clippy.json")
+      .getResources("")
+      .asScala
       .toList
-      .flatMap(loadAdviceFromUrL)
+      .flatMap { entry =>
+        val file = new File(entry.getFile)
+        listFilesForFolder(file)
```

Comment:
I suspect this might not work in a jar - see http://stackoverflow.com/questions/5193786/how-to-use-classloader-getresources-correctly - the answer states that it's not possible to list all resources. So maybe `getResources("")` just works on file-based class loaders, not jar-based? Anyway, one more reason to test :)